src/Integration.UnitTests/Telemetry/TelemetryManagerTests.cs

```diff
@@ -660,14 +628,19 @@ public void SendPayload_DisabledSecretRulesAreUpdated()
 
             telemetryRepositoryMock.Invocations.Clear();
 
-            TriggerTimerElapsed();
+            var payload = new TelemetryPayload();
 
-            telemetryClientMock.Verify(x => x.SendPayloadAsync(
-                It.Is((TelemetryPayload payload) => 
-                    payload.RulesUsage.EnabledByDefaultThatWereDisabled.Count == 2 &&
-                    payload.RulesUsage.EnabledByDefaultThatWereDisabled.Contains("secrets:rule3") &&
-                    payload.RulesUsage.EnabledByDefaultThatWereDisabled.Contains("secrets:rule6"))), Times.Once);
+            payloadCreatorMock.Setup(x => x.Create(
```

Comment:
Small tweak, same logic as before except I'm verifying that it's passed correctly to the payload class instead the client